services/viva-ms/lambdas/decideCaseStatus.js

```diff
@@ -2,37 +2,46 @@
 import to from 'await-to-js';
 
 import config from '../../../config';
+
 import * as dynamoDb from '../../../libs/dynamoDb';
-import { getStatusByType } from '../../../libs/caseStatuses';
 import log from '../../../libs/logs';
+import { getStatusByType, statusTypes } from '../../../libs/caseStatuses';
+
+import putVivaMsEvent from '../helpers/putVivaMsEvent';
 
 export async function main(event, context) {
   const { caseKeys, workflow } = event.detail;
```

Comment:
We should add safe check here so that contents in detail is not undefined.

```JS
function noSuchProperty(key, object) {
  // test if a property doesn't exist in the current object
  return !object.hasOwnProperty(key)
}

if (noSuchProperty('workflow', event.detail)){
  // do something
}

if (noSuchProperty('caseKeys', event.detail)){
  // do something
}

const {caseKeys} = event.detail
 if(noSuchProperty('PK', caseKeys) || noSuchProperty('SK', caseKeys)) {
   // do something
 }

```

Reply:
Or with optional chaining?

```js
if (!event.detail?.workflow) {
  ...
}

if (!event.detail?.caseKeys?.PK || !event.detail?.caseKeys?.SK) {
  ...
}
```

Reply:
Will be fixed in future task